frame/merkle-mountain-range/primitives/src/lib.rs

```diff
@@ -412,4 +516,57 @@ mod tests {
 
 		assert_eq!(decoded_compact, vec![Ok(d.clone()), Ok(d.clone())]);
 	}
+
+	#[test]
+	fn opaque_leaves_should_be_scale_compatible_with_concrete_ones() {
```

Comment:
I had a hard time following this. A few comments indicating the point of each "transformation" would be helpful 😅

Reply:
This should be simplified now, given I've removed the SCALE-compatibility requirement.

Reply:
Yeah, that's a lot easier to follow